File: app_project/backend/app/schemas/news/document.py
from pydantic import BaseModel
from datetime import datetime

class DocumentCategoryBase(BaseModel):
    name: str
    slug: str
    description: str | None = None

class DocumentCategoryCreate(DocumentCategoryBase):
    icon: bytes | None = None

class DocumentCategoryUpdate(DocumentCategoryBase):
    icon: bytes | None = None

class DocumentCategoryResponse(DocumentCategoryBase):
    id: int
    created_at: datetime
    updated_at: datetime
    icon: bytes | None = None

    class Config:
        from_attributes = True


class DocumentBase(BaseModel):
    title: str
    description: str | None = None
    file_url: str | None = None
    content: str | None = None

class DocumentCreate(DocumentBase):
    category_id: int
    image: bytes | None = None

class DocumentUpdate(DocumentBase):
    category_id: int | None = None
    image: bytes | None = None

class DocumentResponse(DocumentBase):
    id: int
    category_id: int
    created_at: datetime
    updated_at: datetime
    image: bytes | None = None

    class Config:
        from_attributes = True

File: app_project/frontend/src/models/news/document.ts
export interface DocumentBase {
  title: string;
  description?: string;
  file_url?: string;
  content?: string;
}

export interface DocumentCreate extends DocumentBase {
  category_id: number;
  image?: File | null;
}

export interface DocumentUpdate extends DocumentBase {
  category_id?: number;
  image?: File | null;
}

export interface DocumentResponse extends DocumentBase {
  id: number;
  category_id: number;
  created_at: string;
  updated_at: string;
  image?: string | null;
}

export interface DocumentCategoryBase {
  name: string;
  slug: string;
  description?: string;
}

export interface DocumentCategoryCreate extends DocumentCategoryBase {
  icon?: File;
}

export interface DocumentCategoryUpdate extends DocumentCategoryBase {
  icon?: File;
}

export interface DocumentCategoryResponse extends DocumentCategoryBase {
  id: number;
  created_at: string;
  updated_at: string;
  icon?: string | null;
}
